main.go

```diff
@@ -50,23 +50,24 @@ import (
-	webhookSvcNamespace = flag.String("webhook-svc-namespace", "spark-operator", "The namespace of the Service for the webhook server.")
-	webhookSvcName      = flag.String("webhook-svc-name", "spark-webhook", "The name of the Service for the webhook server.")
-	webhookPort         = flag.Int("webhook-port", 8080, "Service port of the webhook server.")
-	enableMetrics       = flag.Bool("enable-metrics", false, "Whether to enable the metrics endpoint.")
-	metricsPort         = flag.String("metrics-port", "10254", "Port for the metrics endpoint.")
-	metricsEndpoint     = flag.String("metrics-endpoint", "/metrics", "Metrics endpoint.")
-	metricsPrefix       = flag.String("metrics-prefix", "", "Prefix for the metrics.")
-	ingressUrlFormat    = flag.String("ingress-url-format", "", "Ingress URL format.")
+	master                = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	kubeConfig            = flag.String("kubeConfig", "", "Path to a kube config. Only required if out-of-cluster.")
+	installCRDs           = flag.Bool("install-crds", true, "Whether to install CRDs")
+	controllerThreads     = flag.Int("controller-threads", 10, "Number of worker threads used by the SparkApplication controller.")
+	resyncInterval        = flag.Int("resync-interval", 30, "Informer resync interval in seconds.")
+	namespace             = flag.String("namespace", apiv1.NamespaceAll, "The Kubernetes namespace to manage. Will manage custom resource objects of the managed CRD types for the whole cluster if unset.")
+	enableWebhook         = flag.Bool("enable-webhook", false, "Whether to enable the mutating admission webhook for admitting and patching Spark pods.")
+	webhookConfigName     = flag.String("webhook-config-name", "spark-webhook-config", "The name of the MutatingWebhookConfiguration object to create.")
+	webhookCertDir        = flag.String("webhook-cert-dir", "/etc/webhook-certs", "The directory where x509 certificate and key files are stored.")
+	webhookSvcNamespace   = flag.String("webhook-svc-namespace", "spark-operator", "The namespace of the Service for the webhook server.")
+	webhookSvcName        = flag.String("webhook-svc-name", "spark-webhook", "The name of the Service for the webhook server.")
+	webhookPort           = flag.Int("webhook-port", 8080, "Service port of the webhook server.")
+	enableMetrics         = flag.Bool("enable-metrics", false, "Whether to enable the metrics endpoint.")
+	metricsPort           = flag.String("metrics-port", "10254", "Port for the metrics endpoint.")
+	metricsEndpoint       = flag.String("metrics-endpoint", "/metrics", "Metrics endpoint.")
+	metricsPrefix         = flag.String("metrics-prefix", "", "Prefix for the metrics.")
+	overridePodSecContext = flag.Bool("override-pod-sec-context", false, "Whether to override existing pod security context with patching.")
```

Comment:
I don't think we need a flag to control this. Always overriding as long as users explicitly specify a `PodSecurityContext` in their `SparkApplication`s seems fine to me.

Reply:
Ok sure I had it this way but was not sure if people want more control.